src/sentry/lang/native/symbolizer.py

```diff
@@ -271,36 +271,62 @@ def _convert_symbolserver_match(self, instruction_addr, symbolserver_match, obj)
                 lang=None,
                 symbol=symbol,
-            ), obj, package=symbolserver_match['object_name'])
+            ), package=symbolserver_match['object_name'])
         ]
 
-    def symbolize_frame(self, instruction_addr, sdk_info=None, symbolserver_match=None, trust=None):
-        obj = self.object_lookup.find_object(instruction_addr)
-        if obj is None:
-            if trust == 'scan':
-                return []
-            raise SymbolicationFailed(type=EventError.NATIVE_UNKNOWN_IMAGE)
-
-        # Try to always prefer the images from the application storage.
-        # If the symbolication fails we keep the error for later
+    def symbolize_frame(self, instruction_addr, sdk_info=None,
+                        symbolserver_match=None, symbolicator_match=None,
+                        trust=None):
         app_err = None
-        try:
-            match = self._symbolize_app_frame(instruction_addr, obj, sdk_info=sdk_info, trust=trust)
-            if match:
-                return match
-        except SymbolicationFailed as err:
-            app_err = err
+
+        # A missing symbolicator match indicates that the symbolicator was not
+        # active for this event. Symbolize the app frame directly using
+        # symbolic.
+        # TODO: Remove this after fully switching to symbolicator
+        if symbolicator_match is None:
+            obj = self.object_lookup.find_object(instruction_addr)
+            if obj is None:
+                if trust == 'scan':
+                    return []
+                raise SymbolicationFailed(type=EventError.NATIVE_UNKNOWN_IMAGE)
+
+            # Try to always prefer the images from the application storage.
+            # If the symbolication fails we keep the error for later
+            try:
+                match = self._symbolize_app_frame(
+                    instruction_addr, obj, sdk_info=sdk_info, trust=trust)
+                if match:
+                    return match
+            except SymbolicationFailed as err:
+                app_err = err
+
+        # If the symbolicator was used, trust its result. Errors that were
+        # generated during symbolication are merged into the event's error
+        # array separately and do not need to be handled here. The match
+        # returned can either be:
+        #  - empty: Symbolicator has explicitly discarded this
+        #    frame as a false positive. This happens especially when
+        #    stackwalking without CFI.
+        #  - only unsymbolicated frames:
+        #    Symbolicator was unable to resolve symbols for this frame, so we
+        #    fall back to (iOS) symbolserver (see below).
+        #
+        # TODO: Remove this fallback once symbolicator supports iOS system
+        # symbols and fully trust the symbolicator response.
+        elif any(x["status"] == "symbolicated" for x in symbolicator_match) or symbolicator_match == []:
+            return symbolicator_match
```

Comment:
This will return unsymbolicated frames if symbolication was partially successful. FWIW, this is an unlikely but not entirely impossible case. Mostly, this will happen if the symbolicator resolves inline functions but is then unable to symbolicate them.

Anyway, this needs to be changed so that the returned result will at least not cause the native plugin's `process_frame` to throw up afterwards. 

Also, let's change the `TODO` comment to reflect that only the `symbolserver_match` path is to be removed, but not the (theoretical) `return None` case.

Reply:
discussed IRL that a frame that results in partially symbolicated frames is too much of an edgecase and that we can just discard the entire response as "unsymbolicated", therefore `any` was changed to `all`.